src/string.rs

```diff
@@ -1819,7 +1822,7 @@ impl<A: DeallocRef> String<A> {
     /// # Panics
     ///
     /// Panics if the starting point or end point do not lie on a [`char`]
-    /// boundary, or if they're out of bounds.
+    /// boundary, or if they're out of bounds, or if there is an allocation failure
```

Comment:
Previous undocumented panic condition. `Splice` potentially allocating on `drop` is a very subtle edge case it's super easy to forget!